fix: directories with .sol in the name path treated as files

dbale-altoros opened the pull request:
This PR adds E2E tests for the pr #352 
Credits to @pycckuu

fvictorio commented on e2e/04-dotSol-on-path/contracts/ERC20.sol/Foo.sol.sol:
```diff
@@ -0,0 +1,4 @@
+pragma solidity >=0.5.0;
```
`Foo.sol.sol`?

fvictorio commented on e2e/test.js:
```diff
@@ -111,4 +111,16 @@ describe('e2e', function() {
+  describe('.sol on path', function() {
+    useFixture('04-dotSol-on-path')
+
+    it('should success', function() {
```
```suggestion
    it('should handle directory names that end with .sol', function() {
```

fvictorio commented on CHANGELOG.md:
```diff
@@ -1,12 +1,24 @@
+## [3.3.9] - TBD
```
This change doesn't seem to belong to this PR.

dbale-altoros replied:
I corrected the old changelog and added this pr as well
The point is to have it up to date and update it on every change after this one

fvictorio commented on e2e/test.js:
```diff
@@ -111,4 +111,16 @@ describe('e2e', function() {
+
+      expect(code).to.equal(0)
+
+      console.log(stdout)
```
This console.log doesn't seem necessary.

fvictorio commented on lib/index.js:
```diff
@@ -61,7 +61,7 @@ function processFile(file, config) {
   const ignoreFilter = ignore({ allowRelativePaths: true }).add(config.excludedFiles)
 
-  const allFiles = glob.sync(path, {})
+  const allFiles = glob.sync(path, {}).filter((f) => fs.lstatSync(f).isFile())
```
`glob` has a `nodir` option that we can use instead of doing this.

dbale-altoros commented:
@fvictorio 
I corrected all issues you requested, some of them belong to the original coder.
The changelog is modified because I added this fix and also corrected miss information of previous version
Do you think it can be approved so we can move on ?
Thanks!
